### ISSUE TITLE
[libvirt] error: internal error: argument key 'etcd_prefix' must not have null value

### ISSUE THREAD

**Issue by lnsyyj** [vitalif/vitastor#9]
Hi @vitalif ,

I used vitastor's libvirt patch, and I want to try to use virsh to start the defined virtual machine template through libvirt. The xml is as follows. When I start the virtual machine, the following output appears.

Can you help me take a look?

```
  <devices>
    <emulator>/usr/bin/qemu-system-x86_64</emulator>
      <disk type='network' device='disk'>
        <target dev='vda' bus='virtio' />
        <driver name='qemu' type='raw' />
        <!-- name is Vitastor image name -->
        <!-- config (optional) is the path to Vitastor's configuration file -->
        <!-- query (optional) is Vitastor's etcd_prefix -->
        <source protocol='vitastor' name='vitastor-inode-1' query='/vitastor' config='/etc/vitastor/vitastor.conf'>
          <!-- hosts = etcd addresses -->
          <host name='192.168.3.100' port='2379' />
        </source>
        <!-- required because Vitastor only supports 4k physical sectors -->
        <blockio physical_block_size="4096" logical_block_size="512" />
      </disk>

root@vitastor-1:~/qemuxml# virsh define win10_3
Domain 'vdi_win10_3' defined from win10_3

root@vitastor-1:~/qemuxml# virsh start win10_3
error: Failed to start domain 'win10_3'
error: internal error: argument key 'etcd_prefix' must not have null value
```

**Comment by vitalif** [comment]
Yeah I'll check it. I hit it myself yesterday when I was trying to start an openstack vm with vitastor disk :-) I tested this xml file with libvirt 5.0, but not with 7.0 previously...

**Comment by lnsyyj** [comment]
Thank you, 7.0.0 I will test it for you. O(∩_∩)O

**Comment by lnsyyj** [comment]
I used debian 11, vitastor v0.6.4, qemu 5.2, libvirt 7.0.0, qemu and libvirt used the latest patch of the master branch of vitastor.
Parameter 'inode' is missing.

```
root@vitastor-1:~/qemuxml# virsh create win10_3
error: Failed to create domain from win10_3
error: internal error: process exited while connecting to monitor: qemu-system-x86_64: -blockdev {"driver":"vitastor","etcd_host":"192.168.3.100:2379","etcd_prefix":"/vitastor","image":"vitastor-inode-1","node-name":"libvirt-3-storage","auto-read-only":true,"discard":"unmap"}: Parameter 'inode' is missing
```

**Comment by vitalif** [comment]
Yep I also just encountered this bug. `image` and `config_path` options were absent from QEMU json. I'll update QEMU and Vitastor packages to fix it.

**Comment by vitalif** [comment]
OK, I updated libvirt patches and even added patches for OpenStack and they even work! :-) it's all released as 0.6.5 so you can update and try it out. libvirt is also now prebuilt in the `bullseye` repository

**Comment by lnsyyj** [comment]
Great, I will try.
